capstone-app/src/app/modules/history/wiki-media/wiki-media.component.ts

```diff
@@ -86,48 +83,30 @@ export class WikiMediaComponent implements OnChanges, OnInit {
       .subscribe((result: WikiServiceResult) => {
         this.error = '';
         this.history = result.history;
-        if (!this.history){
-          this.loading = true;
-          this.error = 'Unfortunately, the language you requested does not have an available tranlation '
-          + 'for this page. Please select a different language.';
-        }
-        else {
+        if (this.history.length > 0){
+          this.error = "";
           this.body = this.history;
           this.loading = false;
           this.title = result.title;
           this.urls = result.furtherReading;
+          this.UrlForAttribution = 'https://' + languagePrefix + '.wikipedia.org/wiki/' + queryString;
           if(result.langlinks && result.langlinks.size > 0) {
               this.langlinks = result.langlinks;
           }
         }
-        if (this.history.includes('CPU time usage:')){
-          this.loading = false;
-          this.error = 'Unfortunately, the language you requested has not been fully translated. '
-          + 'The data being shown is what the Wikipedia page returned.';
+        else {
+          this.loading = true;
+          this.UrlForAttribution = '';
+          this.error = 'Unfortunately, the language you requested does not have an available tranlation '
+          + 'for this page. Please select a different language.';
         }
     });
   }
 
   onChangeLanguage(prefix: string) {
-    if (prefix === this.prevPreFix) {
-      this.changeLanguage(this.prevQuery, this.prevPreFix, this.prevWordForHistory);
-    }
-    else if (prefix === 'en' && !this.langlinks.get('en')) {
-      this.prevPreFix = prefix;
-      const language = this.langlinks['ca'];
-      this.query = language['searchQuery'];
-      this.prevQuery = this.query;
-      this.prevWordForHistory = 'History';
-      this.changeLanguage(this.query, 'en', 'History');
-    }
-    else {
       const language = this.langlinks.get(prefix);
-      this.prevPreFix = prefix;
       this.query = language.get('searchQuery');
-      this.prevQuery = this.query;
-      const wordForHistory = this.languages[prefix]['WordForHistory'];
-      this.prevWordForHistory = wordForHistory;
+      const wordForHistory = this.languages[prefix]["WordForHistory"];
```

Comment:
Single quotes

Reply:
Ugh, I always seem to miss that, lol
One day I'll remember.